File: UI/OpticSoftwareUI/src/app/core/entities/user/userDetailEntity.ts
import { UserTypeEnum } from '../../enums/userTypeEnum';
import { BaseEntity } from '../baseEntity';
import { CompanyEntity } from '../company/companyEntity';
import { UserEntity } from './userEntity';

export class UserDetailEntity extends BaseEntity {
  name: string;
  surname: string;
  phoneNumber: string;
  roles: string;
  userType: UserTypeEnum;
  userId: number;
  companyId: number;
  user: UserEntity;
  company: CompanyEntity;
}

File: OpticSoftware.Entity/Entities/User/UserDetailEntity.cs
using OpticSoftware.Entity.Entities.Company;
using OpticSoftware.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace OpticSoftware.Entity.Entities.User
{
    public class UserDetailEntity : BaseEntity
    {
        public string Name { get; set; }
        public string Surname { get; set; }
        public string PhoneNumber { get; set; }
        public string Roles { get; set; }
        public UserTypeEnum UserType { get; set; }


        public long UserID  { get; set; }
        public long CompanyID { get; set; }


        [ForeignKey("UserID")]
        public virtual UserEntity User { get; set; }

        [ForeignKey("CompanyID")]
        public virtual CompanyEntity Company { get; set; }
    }
}
